src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpTypeStyleDiagnosticAnalyzerBase.cs

```diff
@@ -48,11 +48,7 @@ private void HandleVariableDeclaration(SyntaxNodeAnalysisContext context)
             var options = context.Options;
             var syntaxTree = context.Node.SyntaxTree;
             var cancellationToken = context.CancellationToken;
-            var optionSet = options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult();
-            if (optionSet == null)
-            {
-                return;
-            }
+            var optionSet = options.GetAnalyzerOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult();
```

Comment:
It seems that all the call sites for this method have a `.GetAwaiter().GetResult()`. Might be useful to define a `GetAnalyzerOptionSet` helper which does this.

Reply:
Agreed, and especially because once we rip out an remaining dependencies on fetching the Workspace under the covers the async-ness would go away anyways. Better to do it once now.